File: src/main/scala/com/github/mauricio/postgresql/column/BigDecimalDecoder.scala
package com.github.mauricio.postgresql.column

/**
 * User: Maurício Linhares
 * Date: 3/5/12
 * Time: 9:42 AM
 */

object BigDecimalDecoder extends ColumnDecoder {

  def decode( value : String ) : Any = {
    BigDecimal(value)
  }

}

File: mysql-async/src/main/java/com/github/mauricio/async/db/mysql/binary/decoder/BigDecimalDecoder.kt

package com.github.mauricio.async.db.mysql.binary.decoder

import java.nio.charset.Charset
import io.netty.buffer.ByteBuf
import java.math.BigDecimal
import com.github.jasync.sql.db.util.readLengthEncodedString

class BigDecimalDecoder(val charset : Charset ) : BinaryDecoder {
  override fun decode(buffer: ByteBuf): Any {
    return BigDecimal( buffer.readLengthEncodedString(charset) )
  }
}
